luaui/Widgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,277 @@
+end
+
+-- Get a value by uID
+function LRUCache:get(uID)
+    local value = self.cache[uID]
+    if value then
+        -- Move the accessed uID to the end to mark it as most recently used
+        self:moveToEnd(uID)
+        return value
+    else
+        return nil -- uID not found
+    end
+end
+
+-- Put a uID and value into the cache
+function LRUCache:put(uID, value)
+    if self.cache[uID] then
+        -- If uID already exists, just update and mark it as recently used (should never be the case)
+        self.cache[uID] = value
+        self:moveToEnd(uID)
+    end
+
+    -- Check if cache is full, remove oldest item (least recently used)
+    if #self.order >= self.max_size then
+        local lru = table.remove(self.order, 1)
+        self.cache[lru] = nil
+    end
+
+    -- Insert the new uID and its value
+    table.insert(self.order, uID)
+    self.cache[uID] = value
+end
+
+-- Helper function to move uID to the end of the order list
+function LRUCache:moveToEnd(uID)
+    for i, id in ipairs(self.order) do
+        if id == uID then
+            table.remove(self.order, i)
+            break
+        end
+    end
+    table.insert(self.order, uID)
+end
+-- !SECTION LRU Cache
+-- !SECTION OOP
+
+-- SECTION Settings and other variables
+-- constants
+-- DELAY is the number of frames between a given command and getting flag_processed;
+-- Should be bigger than`COMMAND_LIMIT/expected number of nanos`. Default is 15.
+-- NOTE: There is no "overflow" protection. Setting DELAY to low will cause stacks to fill up constantly.
+local DELAY = 15
+-- names of nano turrets names -- TODO Needs Legion nano's too
+local TURRETS = {"armnanotc", "cornanotc", "armnanotct2", "cornanotct2"}
+-- Maximum number of commands to be flag_processed in a single frame; Game sim blocking if too many commands per frame.
+-- Default is 20. But modern hardware could handle more. 20 is sufficient for any game case.
+local COMMAND_LIMIT = 20
+-- variables
+local is_play = false
+local counter = 0
+local flag_listening = false
+local flag_processed = true
+local current_towers = {}
+local command_budget = 0
+local to_be_cleared = {} -- stack of uIDs to be cleared
+local new_orders = {} -- stack of {uid = {cmdArr},...} to be flag_processed
+local radius_cache = LRUCache:new(10) -- LRU cache to reduce the number of calls for model radius
+-- !SECTION Settings and other variables
+
+-- SECTION Helper functions
+-- converts a table to a Set()-like
+local function tableToSet(t) -- coverts a table to a Set()-like
+    for _, key in ipairs(t) do t[key] = true end
+end
+
+-- Returns a float from 2 coordinates
+-- NOTE equal to Spring.GetUnitSeparation(u1, u2, false, false) Uses 3D distance not 2D
+-- This is faster then synch reading from the engine; jumping.
+-- 1.4e changed to use 2D distance x and z instead of 3D, as it seems that nanos work in 2D cylinders.
+local function calculateDistance2D(x1, y1, x2, y2)
+    return math.sqrt((x1 - x2)^2 + (y1 - y2)^2)
+end
```

Comment:
Check if these are already available in the common library, and if not, add them there